pennylane/data/data_manager.py

```diff
@@ -0,0 +1,257 @@
+
+    Returns:
+        list[str]: The paths of files that should be fetched from S3
+    """
+
+    next_folders = folders[1:]
+    folders = set(node) if folders[0] == ["full"] else set(folders[0]).intersection(set(node))
+    if not next_folders:
+        return folders
+    return [
+        os.path.join(folder, child)
+        for folder in folders
+        for child in _generate_folders(node[folder], next_folders)
+    ]
+
+
+def load(
+    data_type, attributes=None, lazy=False, folder_path="", force=False, num_threads=50, **params
+):
+    r"""Downloads the data if it is not already present in the directory and return it to user as a Dataset object
+
+    Args:
+        data_type (str)   : A string representing the type of data required such as `qchem`, `qpsin`, etc.
+        attributes (list) : An optional list to specify individual data element that are required
+        folder_path (str) : Path to the root folder where download takes place.
+            By default dataset folder will be created in the working directory
+        force (Bool)      : Bool representing whether data has to be downloaded even if it is still present
+        num_threads (int) : The maximum number of threads to spawn while downloading files (1 thread per file)
+        params (kwargs)   : Keyword arguments exactly matching the parameters required for the data type.
+            Note that these are not optional
+
+    Returns:
+        list[DatasetFile]
+    """
+
+    _ = lazy
+
+    if data_type not in _foldermap:
+        _refresh_foldermap()
+    if not _data_struct:
+        _refresh_data_struct()
+    if not attributes:
+        attributes = ["full"]
+
+    description = {key: (val if isinstance(val, list) else [val]) for (key, val) in params.items()}
+    _validate_params(data_type, description, attributes)
+    if len(attributes) > 1 and "full" in attributes:
+        attributes = ["full"]
+    for key, val in description.items():
+        if len(val) > 1 and "full" in val:
+            description[key] = ["full"]
+
+    data = _data_struct[data_type]
+    directory_path = os.path.join(folder_path, "datasets")
+
+    folders = [description[param] for param in data["params"]]
+    all_folders = _generate_folders(_foldermap[data_type], folders)
+    _s3_download(data_type, all_folders, attributes, directory_path, force, num_threads)
+
+    data_files = []
+    for folder in all_folders:
+        real_folder = os.path.join(directory_path, data_type, folder)
+        obj = Dataset(data_type, real_folder, folder.replace("/", "_"), standard=True)
+        doc_attrs = obj.list_attributes()
+        doc_vals = [type(getattr(obj, attr)) for attr in doc_attrs]
+        args_idx = [data["attributes"].index(x) for x in doc_attrs]
+        argsdocs = [data["docstrings"][x] for x in args_idx]
+        obj.setdocstr(data["docstr"], doc_attrs, doc_vals, argsdocs)
+        data_files.append(obj)
+
+    return data_files
```

Comment:
Here we return a list when users select a single dataset.
Feels a bit strange to call `dataset[0]` in our examples.
Not sure if it's worth changing, but we can think about it in a future PR.

Reply:
I can be persuaded but tbh I'm not a fan of treating things differently because they happen to return one result, esp if it causes users to add more logic in automation scripts